packages/client/hmi-client/src/components/model-template/tera-model-template-editor.vue

```diff
@@ -352,33 +379,35 @@ function mouseUpdate(event: MouseEvent) {
 	prevY = event.y;
 }
 
-function amrToTemplates() {
-	kernelManager.sendMessage('amr_to_templates', {}).on('amr_to_templates_response', (d) => {
-		// FIXME: Model templates are passed no junctions yet
-
-		// Insert template card data into template models
-		let yPos = 100;
-		const templateModelsWithCards = d.content.templates.map((modelTemplate: any) => {
-			modelTemplate.metadata.templateCard = {
-				id: modelTemplate.header.name,
-				name: modelTemplate.header.name,
-				x: 100,
-				y: yPos
-			} as ModelTemplateCard;
-
-			yPos += 200;
-			return modelTemplate;
-		});
+async function initializeBeakerKernel(isModelLoaded = false) {
+	if (kernelManager.jupyterSession) kernelManager.shutdown();
 
-		templateModelsWithCards.forEach((templateModel: any) =>
-			modelTemplatingService.addCard(decomposedTemplates.value, templateModel)
-		);
-	});
+	try {
+		const context = {
+			context: 'mira_model_edit',
+			language: 'python3',
+			context_info: {
+				id: isModelLoaded ? props.model?.id ?? '' : ''
+			}
+		};
+		await kernelManager.init('beaker_kernel', 'Beaker Kernel', context);
```

Comment:
Something looks a bit off here, it seems like you are opening up far more web-socket connections to beaker than it is necessary? How many do you actually need?

<img width="641" alt="image" src="https://github.com/DARPA-ASKEM/terarium/assets/1220927/0dcdd4fc-d1bd-4390-a790-e49024ebaa97">

Reply:
Now that I put this in tera-model-edit, I'll use the one created there instead of creating a new one. Not exactly sure why there would be more than 2 since I am shutting down the kernel if I want to rerun it. Will look into this

Reply:
Only one session is used within `tera-model-edit` now